src/lib.rs

```diff
@@ -1 +1,452 @@
+use std::{str::FromStr, sync::Mutex};
+
+use crate::sgp4::{
+    Classification, ElementS, EpochToSiderealTimeAlgorithm, Error as SgpError, ErrorTleLine,
+    ErrorTleWhat, Geopotential, NegativeSemiLatusRectum, Orbit, OutOfRangeEccentricity,
+    OutOfRangeEpochEccentricity, OutOfRangePerturbedEccentricity, Prediction, Tle,
+};
+
+use chrono::NaiveDateTime;
+use original::{self};
+
+use wai_bindgen_rust::Handle;
+
 wai_bindgen_rust::export!("sgp4.wai");
+
+struct Sgp4;
+
+impl sgp4::Sgp4 for Sgp4 {
+    fn orbit_from_kozai_elements(
+        geopotential: Geopotential,
+        inclination: f64,
+        right_ascension: f64,
+        eccentricity: f64,
+        argument_of_perigee: f64,
+        mean_anomaly: f64,
+        kozai_mean_motion: f64,
+    ) -> Result<Orbit, SgpError> {
+        match original::Orbit::from_kozai_elements(
+            &geopotential.into(),
+            inclination,
+            right_ascension,
+            eccentricity,
+            argument_of_perigee,
+            mean_anomaly,
+            kozai_mean_motion,
+        ) {
+            Ok(orbit) => Ok(orbit.into()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    fn wgs72() -> Geopotential {
+        original::WGS72.into()
+    }
+
+    fn wgs84() -> Geopotential {
+        original::WGS72.into()
+    }
```

Comment:
Copy pasta ftw :stuck_out_tongue:  

```suggestion
    fn wgs84() -> Geopotential {
        original::WGS84.into()
    }
```